aea/protocols/generator/base.py

```diff
@@ -1921,13 +1923,32 @@ def generate_protobuf_only_mode(self) -> None:
             self.path_to_generated_protocol_package, self.protocol_specification.name
         )
 
-        if is_valid_protobuf_schema:
-            pass
-        else:
+        if run_protolint:
+            self.run_protolint_for_file(
+                os.path.join(
+                    self.path_to_generated_protocol_package,
+                    "{}.proto".format(self.protocol_specification.name),
+                )
+            )
+
+        if not is_valid_protobuf_schema:
             # Remove the generated folder and files
             shutil.rmtree(output_folder)
             raise SyntaxError("Error in the protocol buffer schema code:\n" + msg)
 
+    def run_protolint_for_file(self, filepath: str) -> None:
+        """Perform protolint check for file."""
+        if sys.platform.startswith("win"):
+            protolint_base_cmd = "protolint"
+        else:
+            protolint_base_cmd = "PATH=${PATH}:${GOPATH}/bin/:~/go/bin protolint"
+
+        if call(f"{protolint_base_cmd} version", shell=True) != 0:  # nosec
+            raise ValueError("protolint is not installed!")
+
+        cmd = f'{protolint_base_cmd} lint -fix "{filepath}"'
+        call(cmd, shell=True)  # nosec
```

Comment:
`shell=True` is not recommended for security reason, yet I think we have good control on the `cmd` variable. Maybe some checks on `filepath`? E.g. check that it is actually a path.